### PR TITLE
Web3/projects radspec

### PR DESCRIPTION
**Pull request by stellarmagnet**

### CODE REVIEW

File: apps/projects/contracts/Projects.sol [hunk 1]
```diff
@@ -15,7 +15,7 @@ import "@aragon/os/contracts/lib/math/SafeMath.sol";
   */
 interface Bounties {
     /**
-     * @notice `_sender` will submit a fulfillment for bounty #`_bountyId` on Standard Bounties with the following info: `_data`
+     * @notice `_sender` submit a fulfillment for issue #`_bountyId` with the following info: `_data`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: > This transaction will eventually perform
> 
>   0xb4124cEB3451635DAcedd11767f004d8a28c6eE7 submit a fulfillment for issue #NNNN with the following info: ...
> 
> on 0xblahblahblah

The "_person_ submit a fulfillment" seems like a grammar issue.

File: apps/projects/contracts/Projects.sol [hunk 2]
```diff
@@ -25,7 +25,7 @@ interface Bounties {
     ) external; //{}
 
     /**
-     * @notice `_sender` will update their fulfillment for `_bountyId` on Standard Bounties with the following info: `_data`
+     * @notice `_sender` updated their fulfillment for issue #`_bountyId` with the following info: `_data`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: Should this be past-tense? It's _about to happen_ at the time the user gets this message, right?

File: apps/projects/contracts/Projects.sol [hunk 3]
```diff
@@ -359,8 +363,8 @@ contract Projects is AragonApp, DepositableStorage {
 // Repository functions
 ///////////////////////
     /**
-     * @notice Add repository to the Projects app
-     * @param _repoId id of the repo to be added
+     * @notice Add repository `_repoId`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: I don't think these repo IDs are human-readable, and I doubt it makes sense to put them in the radspec. I think they're just going to look sort of like ugly ethereum-address-style hashes.

File: apps/projects/contracts/Projects.sol [hunk 4]
```diff
@@ -376,7 +380,7 @@ contract Projects is AragonApp, DepositableStorage {
     }
 
     /**
-     * @notice Remove repository from the Projects app
+     * @notice Remove repository `_repoId`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: Same thing here. Do we have some way to display the name of the repo at this point? Would we have to send extra data to the contract, which it wouldn't actually need to use?

File: apps/projects/contracts/Projects.sol [hunk 5]
```diff
@@ -403,7 +407,7 @@ contract Projects is AragonApp, DepositableStorage {
 ///////////////////
 
     /**
-     * @notice Request assignment for issue `_issueNumber`
+     * @notice Request assignment for issue `_issueNumber` with application `_application`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: Is the IPFS hash of their application something the user will recognize (in which case it makes sense to show it here), or something we generate in the background?

File: apps/projects/contracts/Projects.sol [hunk 6]
```diff
@@ -472,7 +476,7 @@ contract Projects is AragonApp, DepositableStorage {
     }
 
     /**
-     * @notice Review work for issue `_issueNumber`
+     * @notice `_approved ? 'Accept' : 'Reject'` work for issue `_issueNumber` with info: `_updatedSubmissionHash`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: Same question here

File: apps/projects/contracts/Projects.sol [hunk 7]
```diff
@@ -673,7 +677,7 @@ contract Projects is AragonApp, DepositableStorage {
     }
 
     /**
-     * @notice Fund issues open to submissions from anyone: `_description`
+     * @notice Fund open submission issues: `_description`
```
Comment (chadoh) [inline apps/projects/contracts/Projects.sol]: Why this change?

File: apps/projects/contracts/Projects.sol (hunk 1 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: Woops. Should be 
> `_sender` submitted a fulfillment for issue #`_bountyId` with the following info: `_data`

File: apps/projects/contracts/Projects.sol (hunk 2 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: Yeah, these are actually user facing and wont be forwarded.

Maybe it should just be
> Update fulfillment for issue #`_bountyId` with the following info: `_data`

File: apps/projects/contracts/Projects.sol (hunk 1 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: Maybe should be:

> Submitted a fulfillment for issue #`_bountyId` with the following info: `_data`

File: apps/projects/contracts/Projects.sol (hunk 3 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: Yeah but there is no other context for the voting app "Add repository" doesn't convey anything. I'm not sure how we can properly solve this, but it's a UX concern if we want people to be able to vote to add or remove a repo. I was hoping maybe we can provide with some kind of tool to look up the repo name via Github app.

File: apps/projects/contracts/Projects.sol (hunk 4 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: I was asking kevin if we can add metadata for the repo @topocount - but if it's IPFS we still have ugly hash until we develop an IPFS radspec resolver

File: apps/projects/contracts/Projects.sol (hunk 5 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: We may have to just put it in the docs so users know it's an ipfs hash. Ideally the client should be able to determine an IPFS hash based on it starting with a Q and it being a certain number of characters, and generate a link that is tied to the client's gateway

Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: For now giving something is better than the users having no data at all in the voting app? i think?

File: apps/projects/contracts/Projects.sol (hunk 7 above)
Comment (stellarmagnet) [reply apps/projects/contracts/Projects.sol]: We are calling it "open submission" in projects UI, and it shortens the text.

Comment (chadoh) [reply apps/projects/contracts/Projects.sol]: Would a hyphen make sense? "Fund open-submission issues"?

File: apps/projects/contracts/Projects.sol (hunk 3 above)
Comment (topocount) [reply apps/projects/contracts/Projects.sol]: I've decided to add a metadata field here, which we can presumable pass the repo name into from the github repo object we have in the frontend

**Comment by stellarmagnet** [comment]
This is great but we will need an updateRepoMetadata action as well so
users can sync new repo names if they change and can also have things like
images associated w the repo. In future when we decentralize this can
manage everything around it like descriptions and other things

**Comment by chadoh** [comment]
Closing; I believe @topocount incorporated these changes directly into #1222 already. Kevin, re-open this PR and correct me if I'm wrong!
